### PR TITLE
#426 implement coordinator notification

### PR DESCRIPTION
**Pull request by aimedivin**
### PR Description
This PR adds the functionality to notify coordinators of important events and updates within the platform.

### Description of tasks  and Functionality
- Added graphql type, query, mutation, and subscription to support notification system.
- Utilizes a global function pushNotification to create a notification and notify coordinator
- Delivers real-time notifications to coordinators using GraphQL subscriptions.
- Ensure that coordinator receive immediate updates on action on platform

### How should this be manually tested?
- Clone this repository and navigate to the project directory
- Ensure that you have set your .env file using .env.example.
```bash
  git clone https://github.com/atlp-rwanda/atlp-pulse-bn.git
  cd atlp-pulse-bn
  git checkout ft-coordinator-notification
  npm i
  npm run dev
  ```

### CODE REVIEW

**Review by Calebgisa72 — APPROVED** [review]

File: src/resolvers/ratingsResolvers.ts [hunk 1]
```diff
@@ -236,35 +237,34 @@ const ratingResolvers: any = {
...
+          //     newRating: {
+          //       id: addNotifications._id,
+          //       receiver: user,
+          //       message: 'Have rated you; check your scores.',
+          //       sender: coordinator,
+          //       read: false,
+          //       createdAt: addNotifications.createdAt,
+          //     },
+          //   })
+          // }
```
Comment (elijahladdie) [inline src/resolvers/ratingsResolvers.ts]: Can your remove this comment if there are no longer needed

File: src/resolvers/ratingsResolvers.ts [hunk 2]
```diff
@@ -579,23 +579,18 @@ const ratingResolvers: any = {
...
-        },
-      })
+      // const addNotifications = await Notification.create({
+      //   receiver: sender?.id == rate?.coordinator ? user : rate?.coordinator,
+      //   message: content,
+      //   sender: sender?.id,
+      //   read: false,
+      //   createdAt: new Date(),
+      // })
+      pushNotification(
```
Comment (elijahladdie) [inline src/resolvers/ratingsResolvers.ts]: remove these comments

File: src/utils/sendEmails.ts [hunk 3]
```diff
@@ -27,12 +28,13 @@ export const sendEmails = async (
     subject: subject,
     html: generateTemplate({ message: content }),
   }
-
+  console.log("2",receiver)
```
Comment (elijahladdie) [inline src/utils/sendEmails.ts]: remove this console

**Review by elijahladdie — CHANGES_REQUESTED** [review]

File: src/resolvers/ratingsResolvers.ts (hunk 2 above)
Comment (aimedivin) [reply src/resolvers/ratingsResolvers.ts]: @elijahladdie Done

**Review by pacifiquemboni — APPROVED** [review]

**Review by Ke-bean — APPROVED** [review]
LGTM

**Review by elijahladdie — APPROVED** [review]
LGTM

**Review by igorihimbazwe — APPROVED** [review]

**Review by MugemaneBertin2001 — APPROVED** [review]
LGTM

**Comment by MugemaneBertin2001** [comment]
but remove these conflicts

**Pull request merged**